src/action/Action.js

```diff
@@ -0,0 +1,50 @@
+export default class Action {
+  constructor(name, character, characters, nodes) {
+    this.character = character;
+    this.characters = characters;
+    this.cooldown = 0;
+    this.name = name;
+    this.nodes = nodes;
+    this.payload = {};
+  }
+
+  getManaCost() {
+    return 0;
+  }
+
+  prepare() {
+    console.log(`${this.character.displayName} will perform ${this.name}...`);
+
+    this._prepare();
+  }
+
+  resolve() {
+    if (this.character.isAlive) {
```

Comment:
Maybe an `isActive` flag would be better, for things like stun. I haven't dug into the world enough to know how we handle that yet, so this might be unnecessary.

Reply:
we should revisit this as we rewrite Action. As I started looking into more complex actions it became pretty clear that the current class will be insufficient. We need to readdress most of this anyway